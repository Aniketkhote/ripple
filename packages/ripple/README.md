<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://raw.githubusercontent.com/trueadm/ripple/master/assets/ripple-dark.png">
  <img src="https://raw.githubusercontent.com/trueadm/ripple/master/assets/ripple-light.png" alt="Ripple - the elegant UI framework for the web" />
</picture>

# What is Ripple?

> Currently, this project is still in early development, and should not be used in production.

Ripple is a TypeScript UI framework for the web. To find out more, view [Ripple's Github README](https://github.com/trueadm/ripple).
