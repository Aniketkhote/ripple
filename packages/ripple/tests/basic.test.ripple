import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { mount } from 'ripple';

describe('basic', () => {
  
  component Basic() {
    <div>{"Hello World"}</div>
  }

  let container;

  function render(component) {
    container = document.createElement('div');
    document.body.appendChild(container);
    mount(component, { target: container });
  }

  beforeEach(() => {
    render(Basic);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it('renders correctly', () => {
    expect(container).toMatchSnapshot();
  });
});
