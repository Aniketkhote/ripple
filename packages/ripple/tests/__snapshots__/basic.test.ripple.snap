// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic > renders correctly 1`] = `
<div>
  <div>
    Hello World
  </div>
  
</div>
`;
